app/Http/Transformers/PostTransformer.php

```diff
@@ -48,11 +47,16 @@ public function transform(Post $post)
                 'total' => $post->reactions_count,
             ],
             'status' => $post->status,
-            'location' => $post->location,
             'created_at' => $post->created_at->toIso8601String(),
             'updated_at' => $post->updated_at->toIso8601String(),
+            'location' => $post->location,
         ];
 
+        // If the post isn't anonymous or if this is the owner of the post, return PII information.
+        if ((! $post->actionModel->anonymous && ! is_null($post->actionModel->anonymous)) || Gate::allows('viewAll', $post) {
```

Comment:
Can you double-check that we're not ending up with an [N+1 query problem](https://secure.phabricator.com/book/phabcontrib/article/n_plus_one/) by querying a property on the `actionModel` relationship here? I think we might want to set the post model to [eager-load](https://laravel.com/docs/5.7/eloquent-relationships#eager-loading) the action relationship [here](https://github.com/DoSomething/rogue/blob/d12bb7f7e9ea622f5672342d53859dbe74b63d80/app/Models/Post.php#L21-L25).

Reply:
ah super smart, it was having the N+1 query problem! Added this!